packages/blade/src/components/Input/OTPInput/__tests__/OTPInput.web.test.tsx

```diff
@@ -95,6 +95,48 @@ describe('<OTPInput />', () => {
     expect(onOTPFilled).toHaveBeenLastCalledWith({ name: 'otp', value: otp });
   });
 
+  it('should handle onFocus', async () => {
+    const label = 'Enter OTP';
+    const onFocus = jest.fn();
+    const user = userEvent.setup();
+    const otp = '123456';
+
+    const { getAllByLabelText } = renderWithTheme(
+      <OTPInput label={label} name="otp" onFocus={onFocus} />,
+    );
+    const allInputs = getAllByLabelText(/character/);
+    await user.type(allInputs[0], otp);
+
+    // should be called for each keystroke
+    expect(onFocus).toHaveBeenCalledTimes(6);
```

Comment:
Why do they want it to be called on every block focus? Shouldn't we call it once? 🤔

Reply:
Each block is a focusable input on its own, if we call it once, which one do we call it for? What if the user focuses on some other top field (like 3rd or 4th) than the one we added the event to? That's why I'm doing the callback for every input and also passing the index of the field that's focused or blurred